lib/Model/Query.js

```diff
@@ -312,7 +317,8 @@ Query.prototype._removeMapIds = function(ids) {
 Query.prototype._addMapIds = function(ids) {
   for (var i = ids.length; i--;) {
     var id = ids[i];
-    this.idMap[id] = true;
+    this.idMap[id] = this.idMap[id] || 0;
+    this.idMap[id]++;
```

Comment:
Equivalently, you can write this: `this.idMap[id] = (this.idMap[id] || 0) + 1;`

It is slightly more complexity on one line, and I don't even know if it makes a difference once optimized, but avoiding two different property gets + assignments might be better.